activerecord/lib/active_record/attribute_methods/primary_key.rb

```diff
@@ -109,14 +80,13 @@ def dangerous_attribute_method?(method_name)
           # Overwriting will negate any effect of the +primary_key_prefix_type+
           # setting, though.
           def primary_key
-            if PRIMARY_KEY_NOT_SET.equal?(@primary_key)
-              @primary_key = reset_primary_key
-            end
+            reset_primary_key if PRIMARY_KEY_NOT_SET.equal?(@primary_key)
```

Comment:
Hey @casperisfine, while trying to make the `pg_party` gem compatible with Rails 7.2 (see https://github.com/rkrage/pg_party/pull/83) I noticed that this diff broke the 7.2 compatibility. 

Is it on purpose that the new version doesn't assign the return value of `reset_primary_key` to `@primary_key` anymore?

Without this assignment the `primary_key` method returns the `BasicObject` value of `PRIMARY_KEY_NOT_SET` which breaks other methods down the line because `BasicObject` doesn't implement most of the comparable methods. 

I saw that `reset_primary_key` calls `primary_key=`, but it seems that it doesn't reflect the `@primary_key` instance variable. Do you have a hunch on how we could solve/fix this?

Reply:
It was on purpose because it seemed useless to me.

> Without this assignment the primary_key method returns the BasicObject value of PRIMARY_KEY_NOT_SET 

Only if `reset_primary_key` doesn't actually reset it. It seems to me you can fix it in your gem: https://github.com/rkrage/pg_party/blob/5acb1e74083945ed327ec905a7b281ad74ece066/lib/pg_party/model/shared_methods.rb#L9

instead of `return xxx`, you can do `return @primary_key = xxx`. Or am I missing something?

Also sorry it broke that gem, but this behavior wasn't tested nor documented, so I don't think it was meant as public API.

Reply:
Oh, you are totally right. I didn't realize it's also overriding `reset_primary_key`. I think I was able to fix it with this commit: https://github.com/rkrage/pg_party/pull/83/commits/65b9d33740421995e2a3cb521f5a6e3f094a19d8 - hoping this is sufficient for the maintainers of `pg_party`.

Also, no need to apologize! Thanks for the quick response! 🙏🏼